drivers: serial: stm32: Enable basic RX/TX FIFO

erwango opened the pull request:
On most STM32, a FIFO mode is available on U(S)ART blocks.
FIFO mode consists in a TX and RX FIFOs, each one 8 frames deep.

By enabling FIFO mode, both FIFOs will be used, using default threshold configuration.
No other change is made to the driver. 
In this case:
- TXE flag stands for TXFNF (TX empty become TXFIFO not full) 
- RXNE flag stands for RXFNE (RX not empty become RX Fifo not empty)
- same for matching interrupts

This has the following benefits:
In TX, FIFO allows to work in burst mode, easing scheduling of loaded applications. It also allows more reliable communication with UART devices sensitive to variation of inter-frames delays.
In RX, FIFO reduces overrun occurences.

erwango committed:
drivers: uart: stm32: Allow enabling FIFO mode

Add required bits to allow FIFO mode enabling.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango committed:
tests: uart: Enable fifo on nucleo_wba2cg for async tests

This will allow compiling/testing fifo in CI.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango committed:
drivers: serial: stm32: Small code refactoring

Wakeup-source configuration is about configuring registers.
It belongs to uart_stm32_registers_configure().

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

reatreada commented on drivers/serial/uart_stm32.c:
```diff
@@ -1957,6 +1957,33 @@ static int uart_stm32_registers_configure(const struct device *dev)
 
+#ifdef USART_CR1_FIFOEN
+	if (config->fifo_enable) {
+		LL_USART_EnableFIFO(config->usart);
```
Maybe it is worth setting `LL_USART_SetTXFIFOThreshold` and ` LL_USART_SetRXFIFOThreshold` explicitly or via DT?

GeorgeCGV commented on drivers/serial/uart_stm32.c:
```diff
@@ -1957,6 +1957,33 @@ static int uart_stm32_registers_configure(const struct device *dev)
 
+#ifdef USART_CR1_FIFOEN
+	if (config->fifo_enable) {
+		LL_USART_EnableFIFO(config->usart);
```
Maybe it is worth setting `LL_USART_SetTXFIFOThreshold` and  `LL_USART_SetRXFIFOThreshold` explicitly or via DT?

erwango replied:
Actually, there is no code to handle thresholds, hence they have no effect.

erwango committed:
drivers: serial: stm32wba: Suspension required before stop in DMA Tx abort

In a previous change, STM32U5 GPDMA specific behavior was set into a
specific configuration applying only to few devices impacted by a specific
silicon erratum.
As part of this change, dma suspension before dma stop was set to apply
to the specific erratum workaround.
It appears, this was wrong and dma suspension before dma stop should
be done on all devices compatible with stm32u5 dma. This fix re-instantiate
the correct behavior.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

GeorgeCGV commented on dts/bindings/serial/st,stm32-uart-base.yaml:
```diff
@@ -86,3 +86,13 @@ properties:
+  fifo-enable:
+    type: boolean
+    description: |
+      Enables transmit and receive FIFO using default FIFO confugration (typically threasholds
+      set to 1/8).
+      In TX, FIFO allows to work in burst mode, easing scheduling of loaded applications. It also
+      allows more reliable communication with UART devices sensitive to variation of inter-frames
+      delays.
+      In RX, FIFO reduces overrun occurences.
```
Minor typos: `configuration`, `thresholds`, `occurrences`.
